modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductDefineExternalyOption.testcase

```diff
@@ -0,0 +1,926 @@
+	@priority = 5
+	test CanViewOnlyValidSKUsAreListedAsExternallyDefinedProducts {
+		property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false";
+		property portal.acceptance = "true";
+
+		task ("Given a business account with a buyer user") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And the U-Joint Product SKU is deleted") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			CommerceNavigator.searchEntry(entryName = "MIN55861");
+
+			CommerceEntry.deleteCommerceEntry(key_value = "MIN55861");
+		}
+
+		task ("And the admin opens the ABS Sensor Product via Admin") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceProducts.searchAndGoToProductEntry(entryProductName = "ABS Sensor");
+		}
+
+		task ("And a new option is created") {
+			CommerceJSONOptionsAPI._addCommerceOption(
+				fieldType = "select",
+				optionKey = "option-select-from-list",
+				optionName = "Option Select from List");
+		}
+
+		task ("And is linked to the product") {
+			CommerceJSONOptionsAPI._assignOptionToCommerceProduct(
+				fieldType = "select",
+				optionKey = "option-select-from-list",
+				optionName = "Option Select from List",
+				productName = "ABS Sensor");
+		}
+
+		task ("When the admin enables the Define Externally toggle with a category") {
+			CommerceEntry.gotoMenuTab(menuTab = "Options");
+
+			CommerceProductOptions.editProductOption(
+				collectionProvider = "SKUs by Categories",
+				key_option = "Option Select from List",
+				key_priceType = "Dynamic",
+				productAddOptionCategoriesList = "Transmission",
+				productOptionDefineExternally = "true",
+				productOptionPriceType = "true");
+		}
+
+		task ("Then all the products related to that category are added except U-Joint") {
+			for (var productEntry : list "CV Axles,Drive Shafts,Mount,Torque Converters,Transmission Cooler Line Assembly,Transmission Fluid,Universal CV Boot,Wheel Seal - Front") {
+				AssertElementPresent(
+					key_entryName = ${productEntry},
+					locator1 = "CommerceEntry#TABLE_ELIGIBILITY_ACCOUNT_GROUPS");
+
+				echo("The product ${productEntry} is present");
+			}
+
+			AssertElementNotPresent(
+				key_entryName = "U-Joint",
+				locator1 = "CommerceEntry#TABLE_ELIGIBILITY_ACCOUNT_GROUPS");
+
+			echo("The product U-Joint is not present");
+		}
+
+		task ("And the buyer can view all the products related to that category are added except U-Joint as options values in product details in the front store") {
```

Comment:
task ("And the buyer can assert that all the products related to that category are added except U-Joint as options values in product details in the front store") {